### ISSUE TITLE
Electrostatic self energies and reciprocal Ewald

### ISSUE THREAD

**Issue by mlund** [mlund/faunus#138]
- [x] Add reciprocal Ewald energy also for Functor and splined potentials, and for dipoles.
- [x] Is Ewald energy added twice in current scheme? The `selfEnergy()` in `CoulombGalore` is added automatically, and so is the self-energy from the `IonIonPolicy`
- [ ] Have tests for self-energies (grand canonical Ewald, qpot?)
- [x] Add self energies for `CoulombGalore` and `DipoleDipoleGalore`
- [x] Complete Ewald documentation, incl. the use of Yukawa Ewald.

ping @bjornstenqvist

**Comment by mlund** [comment]
@gitesei did you have a setup for grand canonical Ewald or q-potential? This could work as tests of the self-energies in the Coulomb pair-potential class.

**Comment by gitesei** [comment]
I have these ones:
https://github.com/gitesei/faunus/blob/master/examples/naclewald1.yml
https://github.com/gitesei/faunus/blob/master/examples/naclewald2.yml
https://github.com/gitesei/faunus/blob/master/examples/naclfanourgakis.yml

**Comment by mlund** [comment]
thanks @gitesei. I will try and add them using the new coulomb energy. I have a branch with updates to the self-energy term. The stockmayer example occasionally fails and on my system it has to do with the debug level being used (-v 3 is ok, but -v 5 is not). The bug is caught by an assertion that hints of something not being transferred from the external CoulombGalore library (ping @bjornstenqvist and @rc83)

**Comment by mlund** [comment]
@bjornstenqvist would you be able to add unittests for `self_energy()` in the coulombgalore project? I realise that we have it only for `plain`. At the moment qPotential in Faunus returns not-a-number.

**Comment by mlund** [comment]
Self energy from above has been merged and there's a new branch, `remove-old-coulomb-galore`, where the old Coulomb galore has been deleted. A few things to do/check:

- [ ] `plain` doesn't take a cutoff but is the only way to use a yukawa potential. How can we add a cutoff?
- [ ] The test `cluster` fails. The initial energy differs slightly from the old CoulombGalore. Could this be due to self-energy, or due to different tolerence on splining in the new CoulombGalore?
- [ ] I have changed the documentation for electrostatics, but this should be expanded with formulas for multipoles as well as for self-energies. Placement of these are already templated in the  md file
